src/Event/EuLoginEventSubscriber.php

```diff
@@ -54,36 +55,69 @@ public static function create(ContainerInterface $container) {
    */
   public static function getSubscribedEvents(): array {
     $events = [];
-    $events[CasHelper::EVENT_PRE_REGISTER] = 'generateEmail';
+    $events[CasHelper::EVENT_POST_LOGIN] = 'updateUserData';
+    $events[CasHelper::EVENT_PRE_REGISTER] = 'generateUserData';
     $events[CasHelper::EVENT_POST_VALIDATE] = 'processAttributes';
     $events[CasHelper::EVENT_PRE_VALIDATE] = 'alterValidationPath';
     return $events;
   }
 
   /**
-   * Generates the user email based on the information taken from EU Login.
+   * Generates the user data based on the information taken from EU Login.
+   *
+   * @param \Drupal\cas\Event\CasPostLoginEvent $event
+   *   The triggered event.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   *   In case of failures an exception is thrown.
+   */
+  public function updateUserData(CasPostLoginEvent $event): void {
+    $properties = $this->generateUserDataArray($event->getCasPropertyBag()->getAttributes());
+    $account = $event->getAccount();
+    foreach ($properties as $name => $value) {
+      $account->set($name, $value);
+    }
+    $account->save();
+  }
+
+  /**
+   * Generates the user data based on the information taken from EU Login.
    *
    * @param \Drupal\cas\Event\CasPreRegisterEvent $event
    *   The triggered event.
    */
-  public function generateEmail(CasPreRegisterEvent $event): void {
+  public function generateUserData(CasPreRegisterEvent $event): void {
     $attributes = $event->getCasPropertyBag()->getAttributes();
+    $event->setPropertyValues($this->generateUserDataArray($attributes));
+  }
+
+  /**
+   * Generates an array of user fields and values based on the CAS attributes.
+   *
+   * @param array $attributes
+   *   An array of attributes retrieved from CAS.
+   *
+   * @return array
+   *   An array of field names and values.
+   */
+  private function generateUserDataArray(array $attributes) {
+    $properties = [];
     if (!empty($attributes['email'])) {
-      $event->setPropertyValue('mail', $attributes['email']);
+      $properties['mail'] = $attributes['email'];
```

Comment:
This is doing a conversion of CAS properties to fields on the user object. This seems like something that can be useful to be reused by other code. Isn't there a better place for this than locked away as an uncallable private method on an event subscriber?